code/datums/components/lightdrawn.dm

```diff
@@ -0,0 +1,136 @@
+		if(C?.glasses)
+			totaltint += C.glasses.tint
+			brightness_mult = min(1, (2 - C.glasses.darkness_view) * 0.33)
+		var/obj/item/organ/eyes/eye = C.getorganslot(ORGAN_SLOT_EYES)
+		if(eye?.tint)
+			totaltint += eye.tint
+		tint_factor = 1/(1 + totaltint)
+
+	var/apparent_brightness = T.lighting_object.cached_max * tint_factor * brightness_mult
+
+	// update their mood
+	if(living_parent.loc != T) // inside something
+		apparent_brightness = 0 // its dark inside things
+		light_mood -= 4
+	else
+		var/change = ((apparent_brightness - light_level_threshold) ** 2)*10
+
+		if(apparent_brightness < light_level_threshold)
+			light_mood -= change
+		else
+			light_mood += change
+
+	light_mood = clamp(light_mood, 0, 100)
+	switch(light_mood)
+		if(0 to light_mood_threshold/2)
+			SEND_SIGNAL(parent, COMSIG_ADD_MOOD_EVENT, "missing_the_light", /datum/mood_event/missing_the_light_a_lot)
+		if(light_mood_threshold/2 to light_mood_threshold)
+			SEND_SIGNAL(parent, COMSIG_ADD_MOOD_EVENT, "missing_the_light", /datum/mood_event/missing_the_light)
+
+	if(light_mood >= light_mood_threshold)
+		if(mesmerised)
+			living_parent.clear_fullscreen("mesmerised")
+			SEND_SIGNAL(living_parent, COMSIG_CLEAR_MOOD_EVENT, "missing_the_light")
+			mesmerised = FALSE
+		return	// mood is okay, done here
+
+	if(apparent_brightness < light_level_threshold)
+		return  // not bright enough, done here
+
+	living_parent.Stun(1 SECONDS)
+
+	if(!mesmerised)
+		living_parent.overlay_fullscreen("mesmerised", /obj/screen/fullscreen/high)
+		SEND_SIGNAL(living_parent, COMSIG_ADD_MOOD_EVENT, "missing_the_light", /datum/mood_event/mesmerised)
+		mesmerised = TRUE
+
+	if(next_move > world.time)
+		return
+
+	var/datum/light_source/brightest
+	var/brightestest = -1
+	for(var/l in T.affecting_lights)
+		var/datum/light_source/light = l
+		if(light.source_atom == living_parent)
+			continue
+		var/brightness = light.light_power * light.light_range
+		var/distance = max(1, get_dist(living_parent, light.source_turf))
+		var/thislightsrelativebrightness = brightness/brightness*sqrt(distance)**2/distance
+
+		if(thislightsrelativebrightness > brightestest)
+			brightest = light
+			brightestest = thislightsrelativebrightness
+
+	if(!brightest)
+		return // no valid light to go to but their mood is still rising
+
+	step_to(living_parent, brightest.source_turf)
+
+	if(next_message < world.time)
+		to_chat(living_parent, "<span class='hypnophrase'>...LÄMP...</span>")
+		next_message = world.time + 8 SECONDS
+
+	next_move = world.time + move_delay
+
+/datum/mood_event/missing_the_light
+	description = "<span class='warning'>I miss the light.</span>\n"
+	mood_change = -2
+	timeout = 5 MINUTES
+
+/datum/mood_event/missing_the_light_a_lot
+	description = "<span class='warning'>I miss the light A LOT.</span>\n"
+	mood_change = -4
+	timeout = 5 MINUTES
+
+/datum/mood_event/mesmerised
+	description = "<span class='warning'>...LÄMP...</span>"
+	mood_change = 10
+	special_screen_obj = "mood_lamp"

```

Comment:
Would be nice if you gave this screen object icon a more generic name, i'm thinking mood_mesmerized